src/renderer/api-decorator.js

```diff
@@ -501,6 +502,41 @@
         return featuresObj;
     }
 
+    function registerWebWindow(name, win, from) {
+        webWindowMap.set(name, win);
+        try {
+            window.opener.fin.__internal_.registerWebWindow(name, win);
+        } catch (err) {
+            //common for main windows, we do not want to expose this error. here just to have a debug target.
+            //console.error(err);
+        }
+    }
+
+    function deregisterWebWindow(name) {
+        webWindowMap.delete(name);
+        try {
+            window.opener.fin.__internal_.deregisterWebWindow(name);
+        } catch (err) {
+            //common for main windows, we do not want to expose this error. here just to have a debug target.
+            //console.error(err);
+        }
+    }
+
+    function getWebWindow(name) {
+        let webWindow = webWindowMap.get(name);
+        if (webWindow) {
+            return webWindow;
+        } else {
+            try {
+                return window.opener.fin.__internal_.getWebWindow(name);
+            } catch (err) {
```

Comment:
Question: If child window is cross domain, window.opener.fin will throw an exception. We're ok with it?

Reply:
yea, the exception is a good improvement over no feedback